File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int zero=0;
        int n=nums.size();
        int left=0;
        int right=0;
        int Max=0;
        while(right<n){
            if(nums[right]==0){zero++;}
            if(zero>1){
                if(nums[left]==0){
                    zero--;
                }
                left++;
            }
            Max=max(Max, right-left);
            right++;
        }
        if (Max==n){return Max-1;}
        return Max;
    }
};

File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: list[int]) -> int:
        zc=0
        l=0
        ans=0
        for i in range(len(nums)):
            if nums[i]==0:
                zc+=1
            while zc>1:
                if nums[l]==0:
                    zc-=1
                l+=1
            ans=max(ans,i-l)
        return ans